### PR TITLE
[RIP-40]Apache Rocketmq Playbook

### PR DESCRIPTION
**Pull request by lwclover**
Apache Rocketmq Playbook可以部署一个完整Apache Rocketmq集群（包括Apache Rocketmq Exporter）。

Apache Rocketmq Playbook具有幂等性，重复执行时不会重启正在运行的进程，所以可以进行Apache Rocketmq集群的在线扩容。

Apache Rocketmq集群缩容涉及到数据的收缩，一般不考虑在Apache Rocketmq Playbook上支持缩容。

Apache Rocketmq Playbook集成了部署环境初始化、可运行包下载、os参数调优、broker最佳配置、Rocketmq集群部署、Exporter部署、添加开机自启动机制等任务编排到一起，实现了大规模集群自动化部署。

### CODE REVIEW

File: rocketmq-ansible/Rocketmq Ansible Playbook cn.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
+Rocketmq playbook提供rocketmq集群部署和rocketmq exporter部署功能。
+
+Rocketmq playbook集成了部署环境初始化、源码包下载、操作系统参数调优、broker最佳配置参数、rocketmq集群部署、rocketmq exporter部署、rocketmq exporter接入、开机自启动等任务编排到一起。
+
+Rocketmq playbook可以嵌入在CI/CD流程中或者编排到terraform流程中，这在自动化运维或者vdc一键部署（SDE）有非常重要的意义。
+
+## 使用说明
+Rocketmq playbook入口文件为rocketmq.yml。rocketmq.yml包含namesrv.yml、broker.yml、exporter.yml3个子playbook。rocketmq.yml可以单独执行部署一个完整的rocketmq集群或者3个子playbook单独执行。hosts文件配置了namesrv、broker、exporter部署的机器列表及变量，当使用terraform编排时hosts可以当做变量传递。
+
```
Comment (duhenglucky) [inline rocketmq-ansible/Rocketmq Ansible Playbook cn.md:9]: 最好加一下 Prerequisite，如何快速上手使用，可以参考下：https://github.com/apache/rocketmq-externals/tree/master/rocketmq-logstash-integration/rocketmq-logstash-input

Comment (lwclover) [reply rocketmq-ansible/Rocketmq Ansible Playbook cn.md:19]: done

File: rocketmq-ansible/roles/broker/templates/logback_broker.xml.j2 [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+  -->
+
+<configuration>
+    <appender name="DefaultAppender"
+              class="ch.qos.logback.core.rolling.RollingFileAppender">
+        <file>{{ rocketmq_data_path }}/log/db-rocketmq/broker_default.log</file>
```
Comment (zhouxinyu) [inline rocketmq-ansible/roles/broker/templates/logback_broker.xml.j2:22]: It's not recommended to modify the path of the log file.

File: rocketmq-ansible/hosts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+127.0.0.2
+
+#rocketmq broker 
+[rocketmq_broker]
+127.0.0.3 brokerName=broker-a brokerId=0 brokerRole=ASYNC_MASTER
+127.0.0.4 brokerName=broker-a brokerId=1 brokerRole=SLAVE
+[rocketmq_broker:vars]
+brokerClusterName=DefaultCluster
+namesrvAddr=127.0.0.1:9876;127.0.0.2:9876
+storePathRootDir=/chj/data/rocketmq/store
```
Comment (zhouxinyu) [inline rocketmq-ansible/hosts:14]: What does the `chj` mean?

**Review by zhouxinyu — CHANGES_REQUESTED** [review]

File: rocketmq-ansible/roles/broker/templates/logback_broker.xml.j2 (hunk 2 above)
Comment (lwclover) [reply rocketmq-ansible/roles/broker/templates/logback_broker.xml.j2:22]: I think the log path still needs to be changed, and the specifications vary from company to company.

Comment (lwclover) [reply rocketmq-ansible/roles/broker/templates/logback_broker.xml.j2]: OK, I will recover it.

File: rocketmq-ansible/hosts (hunk 3 above)
Comment (lwclover) [reply rocketmq-ansible/hosts]: I change it to /root/store

**Review by duhenglucky — APPROVED** [review]

**Review by caigy — CHANGES_REQUESTED** [review]
I couldn't find task about java installation, may it cause startup failure of rocketmq? IMO a `common` role is needed where checking and initialization work can be done.

File: rocketmq-ansible/Rocketmq Ansible Playbook cn.md [hunk 4]
```diff
@@ -0,0 +1,131 @@
+Rocketmq playbook提供rocketmq集群部署和rocketmq exporter部署功能。
```
Comment (RongtongJin) [inline rocketmq-ansible/Rocketmq Ansible Playbook cn.md:1]: 建议写成RocketMQ而不是Rocketmq

Comment (lwclover) [reply rocketmq-ansible/Rocketmq Ansible Playbook cn.md:1]: OK, I will do it soon.

**Comment by lwclover** [comment]
> I couldn't find task about java installation, may it cause startup failure of rocketmq? IMO a `common` role is needed where checking and initialization work can be done.

I will consider it in the next version

**Pull request merged**